clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -201,8 +202,42 @@ public void testSuccessfulHeartbeatTiming() {
     @ParameterizedTest
     @ApiKeyVersionsSource(apiKey = ApiKeys.CONSUMER_GROUP_HEARTBEAT)
     public void testFirstHeartbeatIncludesRequiredInfoToJoinGroupAndGetAssignments(short version) {
-        resetWithZeroHeartbeatInterval(Optional.of(DEFAULT_GROUP_INSTANCE_ID));
+        membershipManager = new MembershipManagerImpl(
+                DEFAULT_GROUP_ID,
+                Optional.of(DEFAULT_GROUP_INSTANCE_ID),
+                0,
+                Optional.of(""),
+                subscriptions,
+                mock(CommitRequestManager.class),
+                (ConsumerMetadata) metadata,
+                logContext,
+                Optional.of(mock(ClientTelemetryReporter.class)),
+                backgroundEventHandler,
+                time,
+                new Metrics()
+        );
+        membershipManager.transitionToJoining();
```

Comment:
actually I wonder if we really need an instance of the `membershipMgr` here (and all the explicit calls on it). Since we're testing how the HBMgr generates the data, we maybe only need to mock the `membershipMgr` funcs the HBMgr relies on, maybe something like this, and removing the membershipMgr instance and all the calls on it, that were only trying to get to the state where these values would be returned:
```
    private void mockJoiningMemberData() {
        when(membershipManager.memberId()).thenReturn("");
        when(membershipManager.memberEpoch()).thenReturn(0);
        when(membershipManager.groupId()).thenReturn(DEFAULT_GROUP_ID);
        when(membershipManager.groupInstanceId()).thenReturn(Optional.of(DEFAULT_GROUP_INSTANCE_ID));
        when(membershipManager.currentAssignment()).thenReturn(LocalAssignment.NONE);
    }
```

Reply:
btw, if this suggestion works, please consider it for the other test on the heartbeat state, maybe we can simplify there too (`testHeartbeatState`)

Reply:
I was able to get this working with this kind of approach, let me know what you think